src/serial.rs

```diff
@@ -1398,388 +1399,449 @@ pub struct Tx<USART> {
             }
+        }
+    }
+}
 
-            impl serial::Write<u8> for Tx<$USARTX> {
-                type Error = Error;
+impl<USART, PINS> blocking::serial::Write<u8> for Serial<USART, PINS>
+where
+    PINS: Pins<USART>,
+    USART: Instance,
+{
+    type Error = Error;
 
-                fn flush(&mut self) -> nb::Result<(), Self::Error> {
-                    // NOTE(unsafe) atomic read with no side effects
-                    let sr = unsafe { (*$USARTX::ptr()).sr.read() };
+    fn bwrite_all(&mut self, bytes: &[u8]) -> Result<(), Self::Error> {
+        let mut tx: Tx<USART> = Tx {
+            _usart: PhantomData,
+        };
+        tx.bwrite_all(bytes)
+    }
 
-                    if sr.tc().bit_is_set() {
-                        Ok(())
-                    } else {
-                        Err(nb::Error::WouldBlock)
-                    }
-                }
+    fn bflush(&mut self) -> Result<(), Self::Error> {
+        let mut tx: Tx<USART> = Tx {
+            _usart: PhantomData,
+        };
+        tx.bflush()
+    }
+}
 
-                fn write(&mut self, byte: u8) -> nb::Result<(), Self::Error> {
-                    // NOTE(unsafe) atomic read with no side effects
-                    let sr = unsafe { (*$USARTX::ptr()).sr.read() };
-
-                    if sr.txe().bit_is_set() {
-                        // NOTE(unsafe) atomic write to stateless register
-                        // NOTE(write_volatile) 8-bit write that's not possible through the svd2rust API
-                        unsafe { ptr::write_volatile(&(*$USARTX::ptr()).dr as *const _ as *mut _, byte) }
-                        Ok(())
-                    } else {
-                        Err(nb::Error::WouldBlock)
-                    }
-                }
-            }
+impl<USART, PINS> Serial<USART, PINS>
+where
+    PINS: Pins<USART>,
+    USART: Instance,
+{
+    fn config_stop(self, config: config::Config) -> Self {
+        unsafe {
+            (*USART::ptr())
+                .cr2
+                .write(|w| w.stop().variant(USART::stopbits(config.stopbits)))
+        };
+        self
+    }
+}
 
-            impl blocking::serial::Write<u8> for Tx<$USARTX> {
-                type Error = Error;
-
-                fn bwrite_all(&mut self, bytes: &[u8]) -> Result<(), Self::Error> {
-                    for &b in bytes {
-                        loop {
-                            match self.write(b) {
-                                Err(nb::Error::WouldBlock) => continue,
-                                Err(nb::Error::Other(err)) => return Err(err),
-                                Ok(()) => break,
-                            }
-                        }
-                    }
-                    Ok(())
+mod private {
+    pub trait Sealed {}
+}
+
+// Implemented by all USART instances
+pub trait Instance: private::Sealed {
+    #[doc(hidden)]
+    fn ptr() -> *const stm32f4::stm32f405::uart4::RegisterBlock;
```

Comment:
This will cause problems, we only have that module when using the f405, maybe try to use `crate::pac` instead.